siege_utilities/geo/providers/etter_filter.py

```
@@ -0,0 +1,293 @@
+            reproducibility — query parsing is not creative writing.
+        api_key: Optional explicit API key; otherwise read from env.
+    """
+    try:
+        from langchain.chat_models import init_chat_model
+    except ImportError:
+        raise ImportError(
+            "default_llm requires langchain. Install with: "
+            "pip install 'siege-utilities[etter]' "
+            "(which pulls langchain via etter), or pass your own "
+            "configured chat model directly to EtterParser."
+        )
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)